users/engler/fairseq/training.py

```diff
@@ -0,0 +1,97 @@
+import os
+import subprocess as sp
+import yaml
+
+from sisyphus import *
+
+import recipe.i6_core.util as util
+
+
+class DictToYamlJob(Job):
+    """
+    Writes a dict into a .yaml file
+    """
+
+    def __init__(self, config_dict, hash_reset, *, yaml_prefix=""):
+        """
+        :param dict config_dict:
+        :param str hash_reset: Temporary way to reset the sisyphus hash for debugging purposes
+        :param str yaml_prefix: Prefix which should be written to the beginning, for example "# @package _group_" or "#!rnn.py"
+        """
+        assert isinstance(config_dict, dict)
+
+        self.config_dict = config_dict
+        self.yaml_prefix = yaml_prefix
+        self.out_config_file = self.output_path("fairseq_config.yaml")
+        self.file_out = self.output_path("out")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        yaml_dict = yaml.dump(self.config_dict)
+        # "# @package _group_" was written at the beginning in the example .yaml from fairseq:
+        if self.yaml_prefix != "":
+            yaml_dict = self.yaml_prefix + "\n" + yaml_dict
+        with open(self.out_config_file.get_path(), "w") as file:
+            file.write(yaml_dict)
+
+
+class FairseqHydraTrainingJob(Job):
+    """
+    Train a Fairseq model using fairseq-hydra-train
+    """
+
+    def __init__(
+        self,
+        python_exe,
+        fairseq_exe,
+        hash_reset,
+        *,  # args below are keyword only
+        additional_args=[],
```

Comment:
Don't use empty lists as a function's default parameter. Use `None` instead and test for it in the function body, e.g., `self.additional_args = additional_args or []`.